src/sentry/snuba/tasks.py

```diff
@@ -114,7 +114,8 @@ def delete_subscription_from_snuba(query_subscription_id, **kwargs):
             QueryDatasets(subscription.snuba_query.dataset), subscription.subscription_id
         )
 
-    subscription.delete()
+    if subscription.status == QuerySubscription.Status.DELETING.value:
```

Comment:
I can't comment on the actual line, but the docstring for this function should be updated to reflect this change.